atest/robot/libdoc/html_output_from_xmlhtml.robot

```diff
@@ -1,117 +1,73 @@
+    ${XML-MODEL}[keywords][0][doc]    ${MODEL}[keywords][0][doc]
+    ${XML-MODEL}[keywords][5][doc]    ${MODEL}[keywords][5][doc]
+    ${XML-MODEL}[keywords][8][doc]    ${MODEL}[keywords][8][doc]
+    ${XML-MODEL}[keywords][11][doc]    ${MODEL}[keywords][11][doc]
 
 Keyword Short Doc
-    ${MODEL}[keywords][1][shortdoc]     A keyword.
-    ${MODEL}[keywords][0][shortdoc]     Get hello.
-    ${MODEL}[keywords][8][shortdoc]     Hyvää yötä.
-    ${MODEL}[keywords][11][shortdoc]    Hyvää yötä.
-
-Keyword Short Doc Spanning Multiple Physical Lines
-    ${MODEL}[keywords][5][shortdoc]    This is short doc. It can span multiple physical lines.
+    ${XML-MODEL}[keywords][1][shortdoc]    ${MODEL}[keywords][1][shortdoc]
+    ${XML-MODEL}[keywords][0][shortdoc]    ${MODEL}[keywords][0][shortdoc]
+    ${XML-MODEL}[keywords][8][shortdoc]    ${MODEL}[keywords][8][shortdoc]
+    ${XML-MODEL}[keywords][11][shortdoc]    ${MODEL}[keywords][11][shortdoc]
+    ${XML-MODEL}[keywords][5][shortdoc]    ${MODEL}[keywords][5][shortdoc]
 
 Keyword tags
-    [Template]    Should Be Equal As Strings
-    ${MODEL}[keywords][1][tags]    []
-    ${MODEL}[keywords][2][tags]    ['1', 'one', 'yksi']
-    ${MODEL}[keywords][3][tags]    ['2', 'kaksi', 'two']
-    ${MODEL}[keywords][4][tags]    ['tag1', 'tag2']
-
-User keyword documentation formatting
-    [Setup]    Run Libdoc And Parse Model From HTML    ${TESTDATADIR}/resource.robot
-    ${MODEL}[keywords][0][doc]    <p>$\{CURDIR}</p>
-    ${MODEL}[keywords][1][doc]    <p><b>DEPRECATED</b> for some reason.</p>
-    ${MODEL}[keywords][2][doc]
-    ${MODEL}[keywords][10][doc]
-    ...    <p>Hyvää yötä.</p>
-    ...    <p>Спасибо!</p>
-    ${MODEL}[keywords][8][doc]
-    ...    <p>foo bar <a href="#kw" class="name">kw</a>.</p>
-    ...    <p>FIRST <span class="name">\${a1}</span> alskdj alskdjlajd askf laskdjf asldkfj alsdkfj alsdkfjasldkfj END</p>
-    ...    <p>SECOND askf laskdjf <i>asldkfj</i> alsdkfj alsdkfjasldkfj askf <b>laskdjf</b> END</p>
-    ...    <p>THIRD asldkfj <a href="#Introduction" class="name">introduction</a> alsdkfj <a href="http://foo.bar">http://foo.bar</a> END</p>
-    ...    <ul>
-    ...    <li>aaa</li>
-    ...    <li>bbb</li>
-    ...    </ul>
-    ...    <hr>
-    ...    <table border="1">
-    ...    <tr>
-    ...    <th>first</th>
-    ...    <th>second</th>
-    ...    </tr>
-    ...    <tr>
-    ...    <td>foo</td>
-    ...    <td>bar</td>
-    ...    </tr>
-    ...    </table>
+    ${XML-MODEL}[keywords][1][tags]    ${MODEL}[keywords][1][tags]
+    ${XML-MODEL}[keywords][2][tags]    ${MODEL}[keywords][2][tags]
+    ${XML-MODEL}[keywords][3][tags]    ${MODEL}[keywords][3][tags]
+    ${XML-MODEL}[keywords][4][tags]    ${MODEL}[keywords][4][tags]
+
+TOC doc
+    [Template]    None
+    Run Libdoc to XML:HTML and to HTML and Parse Models    ${TESTDATADIR}/TOCWithInitsAndKeywords.py
+    Should Be Equal As Strings    ${XML-MODEL}[doc]    ${MODEL}[doc]
```

Comment:
@pekkaklarck 
This is the new test that compares direct model fields with roundtripped model fields.